### PR TITLE
Connect to API

### PR DESCRIPTION
**Pull request by HtetWaiYan7191**
### This pull request includes the following changes : 

- Make use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as your backend.
- Use [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make HTTP requests to the backend
- Fetch books from the Bookstore API after the component has rendered
- Refactor add book and remove book features to persist the changes in the server

Thank you for reviewing my codes 😁

### CODE REVIEW

File: src/components/BooksContainer.js [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
+    dispatch(fetchBook());
+  }, [dispatch]);
+
+  // const fetchBooks = async () => {
+  //   const api = await fetch('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/', { method: 'POST' });
+  //   const result = await api.text();
+  // };
+  // useEffect(() => {
+  //   fetchBooks();
+  // }, []);
```
Comment (Owusu-Desmond) [inline src/components/BooksContainer.js]: - [ ] Please make sure you remove commented code from your project to make your code clean. :pray: You should not push commented code to your repository. You can read more about it [here](https://www.freecodecamp.org/news/code-comments-the-good-the-bad-and-the-ugly-be9cc65fbf83/)

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @HtetWaiYan7191 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

- [ ] Please make sure that you fix this compile problem that comes before the app runs to avoid any issues. :pray: 
  image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BooksContainer.js (hunk 1 above)
Comment (HtetWaiYan7191) [reply src/components/BooksContainer.js]: Hello @Owusu-Desmond , I don't have any compile error when I run it to my local pc. Can you please help me

**Review by levy002 — APPROVED** [review]
## Hello @HtetWaiYan7191 
 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

  unnamed


## Highlights
- No linter error :heavy_check_mark: 
- You implemented all suggested changes :heavy_check_mark: 
 
## Status: APPROVED :fire:  :fire: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**